CommunityShed/CommunityShed/Community.Master.cs

```diff
@@ -13,36 +13,90 @@ namespace CommunityShed
-                int? communityId = CommunityState.GetActiveCommunity();
-                if (communityId != null)
-                {
-                    int personId = CustomPage.CustomUser.PersonId;
+                int personId = CustomPage.CustomUser.PersonId;
 
-                    DataTable communityDataTable = DatabaseHelper.Retrieve(@"
-                    select c.CommunityName
+                DataTable communityDataTable = DatabaseHelper.Retrieve(@"
+                    select c.CommunityName, c.CreatorPersonId
                     from Community c
                     join PersonCommunity pc on pc.CommunityId = c.CommunityId
                     where c.CommunityId = @CommunityId and pc.PersonId = @PersonId
                 ",
-                        new SqlParameter("@CommunityId", communityId.Value),
-                        new SqlParameter("@PersonId", personId));
+                    new SqlParameter("@CommunityId", communityId.Value),
+                    new SqlParameter("@PersonId", personId));
 
-                    if (communityDataTable.Rows.Count == 1)
-                    {
-                        string communityName = communityDataTable.Rows[0].Field<string>("CommunityName");
+                if (communityDataTable.Rows.Count == 0)
+                {
+                    Response.Redirect("~/MyCommunities.aspx");
+                }
 
-                        CommunityNameLabel.Text = communityName;
+                creatorPersonId = communityDataTable.Rows[0].Field<int?>("CreatorPersonId");
 
-                        return;
-                    }
-                }
+                string communityName = communityDataTable.Rows[0].Field<string>("CommunityName");
 
-                Response.Redirect("~/MyCommunities.aspx");
+                CommunityNameLabel.Text = communityName;
+            }
+        }
+
+        protected void Page_PreRender(object sender, EventArgs e)
+        {
+            RenderLeftNav();
+        }
+
+        private void RenderLeftNav()
+        {
+            int personId = CustomPage.CustomUser.PersonId;
+
+            AddNavLink("Available Tools", "Community.aspx");
+            AddNavLink("Borrowed Tools", "BorrowedTools.aspx");
+            AddNavLink("My Tools", "MyTools.aspx");
+            AddNavLink("Add Tool", "ToolAdd.aspx");
+
+            if (CustomPage.CustomUser.IsInRole("Approver"))
+            {
+                AddNavLink("Approve Members", "ApproveMembers.aspx");
             }
+
+            if (CustomPage.CustomUser.IsInRole("Reviewer"))
+            {
+                AddNavLink("Review Disputes", "ReviewDisputes.aspx");
+            }
+
+            if (CustomPage.CustomUser.IsInRole("Enforcer"))
+            {
+                AddNavLink("Members", "CommunityMembers.aspx");
+            }
+
+            // If the current user is the creator of this community
+            // then add an "Edit Community" link.
+            if (personId == creatorPersonId.Value)
```

Comment:
make sure creatorPersonId != null

Reply:
I handled this fix differently. The issue here was that I wasn't retrieving the `creatorPersonId` value on post back. To fix this issue, I updated the `CommunityState` class to store all of the information for the active community, including the `CreatorPersonId` column value.